clients/da-vinci-client/src/main/java/com/linkedin/davinci/kafka/consumer/StoreIngestionTask.java

```diff
@@ -3441,6 +3429,12 @@ private int internalProcessConsumerRecord(
         } catch (Exception e) {
           LOGGER.error("Failed to record Record heartbeat with message: ", e);
         }
+
+        // When Global RT DIV is enabled, VT DIV should be synced to OffsetRecord for non-segment control messages
+        if (isGlobalRtDivEnabled()
+            && shouldSyncOffset(partitionConsumptionState, consumerRecord, leaderProducedRecordContext)) {
+          updateOffsetMetadataAndSyncOffset(partitionConsumptionState);
+        }
```

Comment:
Here is a bit different from the design. For follower, we should have ConsumptionTask to initiate the VT DIV snapshot and pass it to drainer for persistence. Similar to what you have done with `execSyncOffsetFromSnapshotAsync`.